packages/atomic/src/components/insight/user-actions/atomic-insight-user-action/atomic-insight-user-action.tsx

```diff
@@ -0,0 +1,100 @@
+import {Component, h, Prop, State} from '@stencil/core';
+import BookmarkIcon from '../../../../images/bookmark.svg';
+import DocumentIcon from '../../../../images/document.svg';
+import PointIcon from '../../../../images/point.svg';
+import QuickviewIcon from '../../../../images/quickview.svg';
+import SearchIcon from '../../../../images/search.svg';
+import {parseTimestampToDateDetails} from '../../../../utils/date-utils';
+import {InitializeBindings} from '../../../../utils/initialization-utils';
+import {InsightBindings} from '../../atomic-insight-interface/atomic-insight-interface';
+import {UserActionType} from '../atomic-insight-user-action-session/atomic-insight-user-actions-session';
+
+/**
+ * @internal
+ * The `AtomicInsightUserAction` component displays a single user action event in the user action timeline.
+ * @category Insight Panel
+ * @example
+ * <atomic-insight-user-action type="view" actionTitle="atlas data lake" timestamp="1723035731" origin="mySearchHub"></atomic-user-action>
+ */
+@Component({
+  tag: 'atomic-insight-user-action',
+  styleUrl: 'atomic-insight-user-action.pcss',
+  shadow: true,
+})
+export class AtomicInsightUserAction {
+  @InitializeBindings() public bindings!: InsightBindings;
```

Comment:
The AtomicInsightUserAction component will only be used within the `user-actions-session`.

I considered making this component a functional component and I then I didn't cause I read from stencil documentation that functional components do not have scoped styles, while my component needs scooped styles, 
How do you usually style the functional components in Atomic?

For the `user-action-session` is also meant to be used inside the `user-action-timeline` but I didn't make it a functional component because functional components are stateless and the `user-action-session` needs a state.

Reply:
> How do you usually style the functional components in Atomic?
We style them on the top-level class component :) so here, it'd be through `user-action-session`

Reply:
`AtomicInsightUserAction` mad a functional component 👍